src/iter.rs

```diff
@@ -92,3 +93,154 @@ fn hex_chars_to_byte(hi: u8, lo: u8) -> Result<u8, HexToBytesError> {
     let ret = (hih << 4) + loh;
     Ok(ret as u8)
 }
+
+/// Iterator over bytes which encodes the bytes and yields hex characters.
+pub struct BytesToHexIter<I: Iterator<Item = u8>> {
+    /// The iterator whose next byte will be encoded to yield hex characters.
+    iter: I,
+    /// The low character of the pair (high, low) of hex characters encoded per byte.
+    low: Option<char>,
+}
+
+impl<I> BytesToHexIter<I>
+where
+    I: Iterator<Item = u8>,
+{
+    /// Constructs a new `BytesToHexIter` from a byte iterator.
+    pub fn new(iter: I) -> BytesToHexIter<I> { Self { iter, low: None } }
+}
+
+impl<I> Iterator for BytesToHexIter<I>
+where
+    I: Iterator<Item = u8>,
+{
+    type Item = char;
+
+    fn next(&mut self) -> Option<char> {
+        match self.low {
+            Some(c) => {
+                self.low = None;
+                Some(c)
+            }
+            None => self.iter.next().map(|b| {
+                let (high, low) = byte_to_hex_chars(b);
+                self.low = Some(low);
+                high
+            }),
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let (min, max) = self.iter.size_hint();
+        (min * 2, max.map(|max| max * 2))
+    }
+}
+
+impl<I> DoubleEndedIterator for BytesToHexIter<I>
+where
+    I: DoubleEndedIterator + Iterator<Item = u8>,
+{
+    fn next_back(&mut self) -> Option<char> {
+        match self.low {
+            Some(c) => {
+                self.low = None;
+                Some(c)
+            }
+            None => self.iter.next_back().map(|b| {
+                let (high, low) = byte_to_hex_chars(b);
+                self.low = Some(low);
+                high
+            }),
+        }
+    }
+}
+
+impl<I> FusedIterator for BytesToHexIter<I> where
+    I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>
```

Comment:
In fdeb91bf87eb44ea09fe073f57151fd5399363f4:

I don't think we need the `ExactSizeIterator` bound here.

Separately, we should impl `ExactSizeIterator` on this type (and **that** will need an `ExactSizeIter` bound).

Reply:
Yes and yes, done. Thanks.